src/main/java/com/github/pgasync/impl/netty/NettyPgProtocolStream.java

```diff
@@ -250,14 +270,27 @@ public void userEventTriggered(ChannelHandlerContext context, Object evt) throws
         return new ChannelInitializer<Channel>() {
             @Override
             protected void initChannel(Channel channel) throws Exception {
-                if(useSsl) {
+                if (useSsl) {
                     channel.pipeline().addLast(newSslInitiator());
                 }
                 channel.pipeline().addLast(new LengthFieldBasedFrameDecoder(Integer.MAX_VALUE, 1, 4, -4, 0, true));
                 channel.pipeline().addLast(new ByteBufMessageDecoder());
                 channel.pipeline().addLast(new ByteBufMessageEncoder());
                 channel.pipeline().addLast(newProtocolHandler());
                 channel.pipeline().addLast(onActive);
+                channel.pipeline().addLast(newExceptionHandler());
```

Comment:
Why is this moved to a new handler?

Reply:
I wanted to have exception caught regardless of anything and passed to subscriber, also separation of concerns